hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java

```diff
@@ -87,27 +89,27 @@ public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckExcept
         requireNonNull(txn);
         final var op = txn.tokenUpdateOrThrow();
         validateTruePreCheck(op.hasToken(), INVALID_TOKEN_ID);
-
-        if (op.hasFeeScheduleKey()
-                && !isKeyRemoval(op.feeScheduleKey())
-                && op.keyVerificationMode() != TokenKeyValidation.NO_VALIDATION) {
-            validateTruePreCheck(isValid(op.feeScheduleKey()), INVALID_CUSTOM_FEE_SCHEDULE_KEY);
+        // IMPORTANT: No matter the TokenKeyValidation mode, we always require keys to
+        // be structurally valid. Putting structurally invalid keys into ledger state
+        // makes no sense, and could create problems for mirror nodes and block explorers.
+        // That is, using NO_VALIDATION only lets a user set a new key without its signature;
+        // and thus set a low-entropy key that proves a role function has been "disabled"
+        for (final var tokenKey : TOKEN_KEYS) {
+            if (tokenKey.isPresentInUpdate(op)) {
+                final var key = tokenKey.getFromUpdate(op);
+                if (!isKeyRemoval(key)) {
```

Comment:
```suggestion
                if (!isKeyRemoval(key) && op.keyVerificationMode() != TokenKeyValidation.NO_VALIDATION) {
```
if we do not check for the `TokenKeyValidation` enum, a user will always end up with `INVALID_<SOME>_KEY` in prechecks when he tries to update to an invalid key. IMO, we should check if the `TokenKeyValidation` is `NO_VALIDATION` and in this case to not throw anything and allow the user to update to an invalid key

Reply:
I have opened a [HIP PR](https://github.com/hashgraph/hedera-improvement-proposal/pull/950) to eliminate misleading use of the term "invalid"; we definitely don't want to introduce a way to put invalid key structures in state.